### ISSUE TITLE
How are underlines positioned in Ethiopic text?

### ISSUE THREAD

**Issue by r12a** [w3c/elreq#110]
CSS and browser implementers are looking for information about how to position underlines for various scripts, including Ethiopic. Can we provide them with the information they need (and add it to the layout requirements doc)? See w3c/csswg-drafts#459 (comment)

**Comment by dyacob** [comment]
Yes, this would be good to work out.  Underline for Ethiopic lends itself to an optically balanced approach where the underline thickness varies with the font weight (stroke width). This is evident in corpus and an (underline / stroke-width) ratio can be calculated.

One issue to raise for Ethiopic underline occurs when the underline passes below an Ethiopic numeral (which contains a stroke below the baseline).  The resulting overlap doesn't look good. The gap as shown in the discussion for Arabic underline (2nd message) would work well as a solution here. I can't find a way to reference the message here, but the image link is the following:

https://cloud.githubusercontent.com/assets/531910/20427559/e090f62c-ad99-11e6-8740-414c0e28caf6.png

**Comment by moyogo** [comment]
The comment referred to by @dyacob is https://github.com/w3c/alreq/issues/86#issuecomment-261503884.

@dyacob Click on the date at the top for the direct link to a comment, “3 days agao” in “mostafah commented 3 days ago”.

**Comment by r12a** [comment]
@dyacob hth: this question was particularly about how to determine the position of an underline relative to the text, but your comment mentions two other topics related to underlines: (a) how to get the right thickness, and (b) how to skip parts of characters that would interfere with the underline. Here are some relevant CSS links:

(a)  https://github.com/w3c/csswg-drafts/issues/239 (and parts of https://github.com/w3c/csswg-drafts/issues/459)

(b) https://drafts.csswg.org/css-text-decor-3/#text-decoration-skip-property - there are also some issues at https://github.com/w3c/csswg-drafts/issues?utf8=%E2%9C%93&q=skip

I'd suggest that the link at the top of the page https://github.com/w3c/csswg-drafts/issues/459 is probably best restricted to discussion of the distance of the underline from the text.

**Comment by dyacob** [comment]
@moyogo thanks for the tip, very helpful.  @r12a thanks for the references.

**Comment by DMequanint** [comment]
@dyacob thanks for the precise info. provided. Just to add my take in the thread, two issues stand out:
1. Weight of the underline needs to be proportional with the weight of the font.
2.  For texts that involve Ethiopic numerals the underline interferes with the base of the numeral
![image](https://cloud.githubusercontent.com/assets/25501395/22543815/3e598d8e-e943-11e6-9415-2c3e42f23d70.png)

**Comment by dyacob** [comment]
@DMequanint , yes you have correctly captured the two issue for consideration.  Both are sensitive to the type design.  The 2nd issue is, in general, expected but may not occur depending on how the type designer has created the numerals.

On the first issue, I don't want to over state it.  In most 20th century literature, pre-software based publishing, the heavier underline stroke seen with Ethiopic text may be just be an artifact of the printing equipment more than it is a conscious design decision by someone.

Ethiopic text can be formatted satisfactorily with "the default" underline that I've observed in word processors and web browsers, that is not proportional to typeface weight.  A thick or semi-thick underline should also be available for authors who favor the heavier aesthetic.
